packages/flutter/test/cupertino/switch_test.dart

```diff
@@ -414,5 +414,39 @@ void main() {
 
     expect(value, isFalse);
   });
+  testWidgets('Switch is translucent when disabled', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      const Directionality(
+        textDirection: TextDirection.ltr,
+        child: Center(
+          child: CupertinoSwitch(
+            value: false,
+            dragStartBehavior: DragStartBehavior.down,
+            onChanged: null,
+          ),
+        )
+      ),
+    );
+
+    expect(find.byType(Opacity), findsOneWidget);
+    expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 0.5);
+  });
+  testWidgets('Switch is opaque when enabled', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Center(
+          child: CupertinoSwitch(
+            value: false,
+            dragStartBehavior: DragStartBehavior.down,
+            onChanged: (bool newValue) {},
+          ),
+        )
+      ),
+    );
+
+    expect(find.byType(Opacity), findsOneWidget);
+    expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 1.0);
+  });
 
```

Comment:
Thanks for the tests. Looks good. Can you add one more test to check that going from enable to disabled and going from disabled to enabled take effect as you change?

Reply:
I'm having trouble achieving this, currently I'm trying to do something like this:

`
  testWidgets('Switch becomes opaque once enabled', (WidgetTester tester) async {
    bool isEnabled = false;

    await tester.pumpWidget(
      Directionality(
        textDirection: TextDirection.rtl,
        child: StatefulBuilder(
          builder: (BuildContext context, StateSetter setState) {
            return Center(
              child: CupertinoSwitch(
                value: false,
                dragStartBehavior: DragStartBehavior.down,
                onChanged: isEnabled ? (bool newValue) {} : null,
              ),
            );
          },
        ),
      ),      
    );

    expect(find.byType(Opacity), findsOneWidget);
    expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 0.5);

    isEnabled = true;
    await tester.pump();

    expect(find.byType(Opacity), findsOneWidget);
    expect(tester.widget<Opacity>(find.byType(Opacity).first).opacity, 1.0);
  });
`
It's not working because the widget is not rebuilding with the new value. I can't seem to find any examples of this being done in the other test files. Do you know how I can do this or am I taking the wrong approach?

Reply:
To get the effect you desire, you should just be able to declare the same tree again, within another call to `tester.pumpWidget()`, but in the 2nd tree you should change the `onChanged` property to be non-null. Does that make sense?

Reply:
Yes I saw that in some other test files but I didn't think i would create the effect I needed. But after further thought it makes sense, thanks for you help